.github/workflows/release.yaml

```
@@ -0,0 +1,66 @@
+---
+name: Release
+
+run-name: release ${{ inputs.tag }}
+
+on:
+  workflow_dispatch:
```

should we enfore branch name as well to prevent releasing from other branch than dev ?

```suggestion
  workflow_dispatch:
    branches:
      - development/*
      - hotfix/*
```

---

I'm not sure it works with workflow_dispatch

---

Or maybe something like
```yaml
if: github.ref == 'refs/heads/master'
```

or 
```yaml
# Only run for protected branches
if: github.ref_protected == true
```

to avoid releasing from unprotected branch something that was not review

---

Yeah not sure the `branches` at the `workflow_dispatch` level works, but we can add a `if` condition at the step or job level for sure.

If we want actual protection, it can be done via `environments`

---

`if` will skip the job, I think it would actually be better to _fail_ the job is the branch is not correct: i.e. just add another sanity check step at the begining...

(similarly, there are actually quite a few other check that may be done: i.e. is the tag not already released, does the tag match the branch, ...)

---

I added a couple of checks, but if it grows bigger than that I suggest we make it a topic, because then comes questions of making some kind of actions for it, to share it around, etc etc.

---

For the tag, shouldn't the action fail to push if it already exists ?